src/Cli/Abstractions/Reporter.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Note that this file is copied from: https://github.com/dotnet/sdk
```

Comment:
For all of these references, please link to the exact code file and use a permalink